### ISSUE TITLE
Java CDKTF unusual behavior when running `cdk provider add` twice.

### ISSUE THREAD

**Issue by Zman94** [hashicorp/terraform-cdk#1994]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### cdktf & Language Versions

```
❯ cdktf provider add "aws@~>4.0"
Checking whether pre-built provider exists for the following constraints:
  provider: aws
  version : ~>4.0
  language: java
  cdktf   : 0.12.0
```
### Affected Provider / Resource(s)

AWS and Azurerm

### Debug Output

https://gist.github.com/Zman94/5963ed0a2c1a1668db821c89922cfc16#file-terminal-output

### Expected Behavior

Every run of `cdktf provider add "aws@~>4.0"` should use cdktf version 0.12.0

### Actual Behavior

The first run uses:
```
  cdktf   : 0.12.0
```

The second run uses:
```
  cdktf   :  com.hashicorp:cdktf-provider-aws:jar:9.0.9
```
And proceeds to download the entire AWS provider which then causes synthesizing to take 10+ minutes

### Steps to Reproduce

See gist.
TLDR
```
❯ mkdir learn-cdktf
❯ cd learn-cdktf
❯ cdktf init --template=java
❯ cdktf provider add "aws@~>4.0"
❯ cdktf provider add "aws@~>4.0"
```

**Comment by ansgarm** [comment]
Thank you for reporting this @Zman94!

I can reproduce it. After installing the pre-built provider, our detection for the current CDKTF versions seems to break somehow (by picking up the provider package).
The following command also shows this:
```
# cdktf debug
cdktf debug
language: java
cdktf-cli: 0.12.0
node: v14.18.3
cdktf:  com.hashicorp:cdktf-provider-aws:jar:9.0.9 # 👀
constructs: 10.0.25
jsii: 1.63.2
terraform: 1.0.7
(...)
```

**Comment by Zman94** [comment]
You guys rock! Thanks so much!
